sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1543,4 +1543,27 @@ class AdaptiveQueryExecSuite
     assert(materializeLogs(0).startsWith("Materialize query stage BroadcastQueryStageExec"))
     assert(materializeLogs(1).startsWith("Materialize query stage ShuffleQueryStageExec"))
   }
+
+  test("SPARK-34899: Use origin plan if we can not coalesce shuffle partition") {
+    def check(ds: Dataset[Row], origin: ShuffleOrigin): Unit = {
```

Comment:
`check` -> `checkNoCoalescePartitions`?

Reply:
Also, how about checking the initial partition num, too?

Reply:
Added the test